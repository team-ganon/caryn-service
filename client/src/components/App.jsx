import React from 'react';
import ReactDOM from 'react-dom';
import Introduction from './Introduction.jsx';
import $ from 'jquery';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      intro: []
    }
    this.getPhotos = this.getPhotos.bind(this);
  }

  getPhotos () {
    $.ajax({
      method: 'GET',
      url: `http://localhost:3002/api/intro`,
      success: function(data) {
        // console.log('in success')
        this.setState({
          intro: data
        })
      }.bind(this)
    })
  }

  componentDidMount () {
    // var endPoint = window.location.href.split('api/');
    this.getPhotos();
  }
  render () {
    return (
      <div>
        <Introduction info={this.state.intro}/>
      </div>
    )
  }
}

// ReactDOM.render(<App />, document.getElementById('app'));

export {App};