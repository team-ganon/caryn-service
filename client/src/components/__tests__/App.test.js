import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { App } from '../App.jsx'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


describe('App', () => {
  it('see if jest works', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot();
  })
})


