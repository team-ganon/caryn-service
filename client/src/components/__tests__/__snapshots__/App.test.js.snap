// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App see if jest works 1`] = `
<div>
  <Introduction
    info={Array []}
  />
</div>
`;
